core/src/main/java/org/elasticsearch/search/aggregations/pipeline/movavg/models/LinearModel.java

```diff
@@ -41,6 +42,29 @@
     protected static final ParseField NAME_FIELD = new ParseField("linear");
 
     @Override
+    public <T extends Number> double[] predict(Collection<T> values, int numPredictions) {
```

Comment:
in fact, to make all the models work in the same way, we could make MovAvgModel declare the following abstract method:
```java
 public abstract <T extends Number> double[] next(Collection<T> values, int numForecasts);
```
Then the predict and next methods can just be declared in MovAvgModel and be the following:
```java
    public final <T extends Number> double[] predict(Collection<T> values, int numPredictions) {
        return next(values, numPredictions);
    }

    public final <T extends Number> double next(Collection<T> values) {
        return next(values, 1)[0];
    }
```